### DISCUSSION TITLE
How to get the calculated embedding for each segment from the pipeline output ?

### DISCUSSION THREAD

**Discussion by amitli1** [pyannote/pyannote-audio#1338] [Development]
Is there a way to get the calculated embedding for each segment from the pipeline result ?

```
from pyannote.audio                    import Pipeline
pipeline   = Pipeline.from_pretrained("pyannote/speaker-diarization")
diarization = pipeline("test.wav")
# how to get the embedding for each segment from diarization ?
```

How can I get the caluclated embedding for each segement ? 

I know I can iterate over (`diarization.itertracks`) and get the relevant segment times and recalculate the embedding (`pyannote/embedding`), but it seems redundant because this information has already been calculated.

**Comment by KarimMibrahim** [comment]
Did you find a solution?

**Reply by amitli1** [reply]
No

**Comment by hbredin** [comment]
The solution you (@amitli1) propose is actually the only one I can think of.

Indeed, separate embeddings for each segment are in fact never computed during inference.
I suggest you read [this paper](https://www.isca-speech.org/archive/interspeech_2023/bredin23_interspeech.html) (and more precisely Section 2.2) for a better understanding of how the pipeline works.
